### DISCUSSION TITLE
Discussion API bugs and feature requests

### DISCUSSION THREAD

**Discussion by laymonage** [community/community#15435] [Discussions]
Hey folks, awesome work you've been doing on the Discussions feature so far. I noticed it's been a while since we have updates to the Discussion API. I use the API extensively for my app [**giscus**](https://github.com/giscus/giscus), and I've been waiting for some updates to the API so I can implement new features on my app.

Sadly, this Discussions Feedback section seems to have been filled with spam and unrelated discussions lately, which makes bug reports and feature requests less visible. I've decided to create this discussion to list the fixes and new features I'd like to be available in the API, so that it's easier for you to see:

Ordered from most wanted.
- #3968
- #8853
- #15432
- #15431
- #4113

Some of these are almost a year old. It would be really nice to have them in the API soon.

Nice to have:
- #36
- That, and/or:
  - Allow creating discussions and comments as a Mannequin user and with a custom timestamp. I know this is something that's possible (like what you did with [Python's recent issues migration](https://github.com/python/cpython/issues?q=is%3Aissue+is%3Aopen+created%3A%3C2022-01-01+)), but I guess this isn't something that's open for the general public? It'd be nice to have this, though. User can come up with their own solution to convert the data from any other platform into something that's compatible with GitHub Discussions, then use the API to create the data on GitHub.
- API for the [Polls feature](https://github.blog/2022-04-12-whats-new-in-github-discussions-organization-discussions-polls-and-more/) (not necessary for my use case, but would be useful for others)

I will update this discussion if I find anything else. Thanks!

**Comment by laymonage** [comment]
Tagging @evi-liu, I hope you don't mind!

**Comment by laymonage** [comment]
Bump. This discussion category is filled with offtopic and spam these days, making it hard to see actual feedback 😢

**Comment by laymonage** [comment]
Bump.

**Comment by Bestra** [comment]
@laymonage thanks so much for the extra bumps on this! I'll respond individually on each issue but I'll also summarize here. I'm hoping we can get these prioritized relatively soon (way sooner than the amount you've waited already 😅 )

-   [Cannot call addUpvote/removeUpvote mutations using app-issued user token #3968](https://github.com/orgs/community/discussions/3968)
    -   I found the issue for this from last year and reopened it, I'll bring this up for triaging.
-   [[BUG?] Searching private discussions using the GraphQL API returns no results when using app-issued token #8853](https://github.com/orgs/community/discussions/8853)
    - I see we made an issue for this about a month ago but we haven't picked it up yet.
-   [Discussions API: Allow specifying order in DiscussionCommentConnection #15432](https://github.com/orgs/community/discussions/15432)
    -   I wrote up a new issue for this. We actually have almost exactly what you want already, but it's only exposed internally at the moment. I'm hoping we can turn this one around quickly but we'll see. 
-   [Discussions API: Allow specifying label in createDiscussion and updateDiscussion mutations #15431](https://github.com/orgs/community/discussions/15431)
    -  I wrote up a new issue for this one just now.
-   [Categories in search query should be case-insensitive #4113](https://github.com/orgs/community/discussions/4113)
    -   Thanks for the excellent writeup! I've put an issue into our backlog.

**Comment by seanbudd** [comment]
Adding and API for the polls feature would be great, we'd like to use this in nvaccess/nvda
